### ISSUE TITLE
Move into ruby-shell organization?

### ISSUE THREAD

**Issue by lostinblue** [s-mage/rush#22]
To facilitate community development of this project would you consider migrating it to a ruby-shell organization that multiple developers could be added too?

**Comment by s-mage** [comment]
Only if anyone uses it :) I have a strong feeling that I'm the only one who uses it as a primary shell

**Comment by lostinblue** [comment]
I have switched, the first thing I noticed was I need to do vim "file.rb", do you think this is solvable? 

It would also be nice to do

    p "thing > file

So p/print can replace echo.

**Comment by lostinblue** [comment]
I have also been looking at building/using a ruby interpreter built in a compiled language and implement rush in another faster language but still use ruby to interact with unix. 

I think the advantage would be speed, ability to compile into a binary. 

What is interesting is I had the same idea when sketching up a unix replacement with the clustering shell. This project has a lot of potential.

**Comment by lostinblue** [comment]
"I overload method_missing to shell out if a command exists, or to return a string otherwise."

Would this be a reasonable thing to do? Do you have an opinion on it? I think it would fix my vim "file.rb" issue, which makes my work a bit more clunky or atleast my adjustment.

**Comment by s-mage** [comment]
@lostinblue first of all, 

  vim "file.rb"

this would work, but this is not intended to work like this. I already overloaded method_missing to run commands but it's kind of fallback.

  home['file.rb'].vim

This is how it should work.

  p 'thing' > file

Again, there is another way to do this

  file.write 'thing'

About rewriting rush in another language -- I wouldn't do it. You can try, but for me it's not worth it.

**Comment by drosehn** [comment]
This is an interesting idea, although I'm not sure that it'd work well enough for me to use it in place of bash.

If you were looking for a ruby-like language which is compiled, you could check out [crystal](https://crystal-lang.org) .  However, crystal is still in development and I expect there are many things you'd want to do when writing a shell where crystal doesn't have the support written yet.  I also expect it would be a **lot** of work to switch this to be any compiled language.

**Comment by drosehn** [comment]
Getting back to `rush` itself, I do drop into `irb` quite often because I work a lot with ruby so some things are just easier for me to do in `irb` than in `bash`.  But the way `rush` is set up is (IMO) too verbose for 90% of the commands that I do in `bash` (and I do spend a lot of time working in the unix command-line).

So I like the idea of `rush`, but for it to catch on I think that at least the simpler commands need to be more like standard `sh`/`bash` shell commands.

**Comment by s-mage** [comment]
@drosehn 
> But the way rush is set up is (IMO) too verbose for 90% of the commands that I do in bash

Yes, after some time I can't ignore that it's a problem. I don't know how to improve it. For myself I've added a lot of aliases. Part of my config:

```
def nvim(*args)
  if self.class == Rush::Dir
    system "cd #{full_path}; nvim +Files"
  else
    super
  end
end

alias_method :v, :nvim

def matlab; open_with :matlab, '-nodesktop -nosplash'; end
def scilab; open_with 'scilab-adv-cli'; end

def sgem(*args); open_with :gem, *args; end

def rs(*args); rails :s, *args; end
def rc(*args); rails :c, *args; end
def rg(*args); rails :g, *args; end

def s(*args); rspec *args; end

def rr; output_of :bundle, :exec, :rake, :routes; end
def migrate(*args); output_of :bundle, :exec, :rake, 'db:migrate', *args; end

# bundle aliases.
def bi(*args); bundle :install, *args; end
def bu(*args); bundle :update,  *args; end
def be(*args); bundle :exec,    *args; end

# git aliases.
def g(*args);   git          *args; end
def gb(*args);  g :branch,   *args; end
def gp(*args);  g :pull,     *args; end
def gps(*args); g :push,     *args; end
def gad(*args); g :add,      *args; end
def gs(*args);  g :status,   *args; end
def gd(*args);  g :diff,     *args; end
def gm(*args);  g :merge,    *args; end
def gco(*args); g :checkout, *args; end
def gc(*args)
  if Integer === args.first && String === args[1]
    git :commit, "[refs ##{args.first}] #{args[1]}"
  elsif args.empty?
    open_with :git, :commit
  else
    git :commit,   *args
  end
end

# git checkout to popular branches
def gcom(*args); g :checkout, :master, *args; end
def gcod(*args); g :checkout, :develop, *args; end
def gcop(*args); g :checkout, :-, *args; end # previous branch


# git flow
def gf(*args);   g :flow,     *args; end
def gff(*args);  gf :feature, *args; end

# lein aliases
def lr(*args); lein :repl,     *args; end
def lt(*args); lein :test,     *args; end
def lf(*args); lein :figwheel, *args; end
```
